backend/app/app/api/api_v1/endpoints/monitoringevent.py

```diff
@@ -1,179 +1,338 @@
-        
-        #Create the reference resource and location header
-        link = str(http_request.url) + '/' + str(inserted_doc.inserted_id)
-        response_header = {"location" : link}
-        
-        #Update the subscription with the new resource (link) and return the response (+response header)
-        crud_mongo.update_new_field(db_mongo, db_collection, inserted_doc.inserted_id, {"link" : link})
-
-        #Retrieve the updated document | UpdateResult is not a dict
-        updated_doc = crud_mongo.read_uuid(db_mongo, db_collection, inserted_doc.inserted_id)
-        updated_doc.pop("owner_id")
+    if item_in.self is not None:
+        raise HTTPException(
+            status_code=400, detail="The self attribute must not be set"
+        )
 
+    db_mongo = client.fastapi
 
-        http_response = JSONResponse(content=updated_doc, status_code=201, headers=response_header)
-        add_notifications(http_request, http_response, False)
-        
-        return http_response
-    elif (item_in.monitoringType == "LOSS_OF_CONNECTIVITY" or item_in.monitoringType == "UE_REACHABILITY") and item_in.maximumNumberOfReports == 1:
-        return JSONResponse(content=jsonable_encoder(
-            {
-                "title" : "The requested parameters are out of range",
-                "invalidParams" : {
-                    "param" : "maximumNumberOfReports",
-                    "reason" : "\"maximumNumberOfReports\" should be greater than 1 in case of LOSS_OF_CONNECTIVITY event"
+    ue = None
+
+    id = ObjectId()
+    item_in.self = parse_obj_as(Link, f"{http_request.url}/{id}")
+
+    if item_in.ipv4Addr is not None:
+        id_value = str(item_in.ueIpv4Addr)
+        ue = crud_ue.get_ipv4(db=db, ipv4=id_value, owner_id=current_user.id)
+
+    elif item_in.ipv6Addr is not None:
+        id_value = item_in.ueIpv6Addr.exploded
+        ue = crud_ue.get_ipv6(db=db, ipv6=id_value, owner_id=current_user.id)
+
+    elif item_in.externalId:
+        ue = crud_ue.get_externalId(
+            db=db, externalId=item_in.externalId, owner_id=current_user.id
+        )
+
+    elif item_in.msisdn:
+        ue = crud_ue.get_msisdn(
+            db=db, msisdn=item_in.msisdn.__root__, owner_id=current_user.id
+        )
+
+    if not ue:
+        raise HTTPException(
+            status_code=404, detail="UE with this identifier doesn't exist"
+        )
+
+    isByMaxReports = item_in.maximumNumberOfReports
+
+    isByExpireTime = item_in.monitorExpireTime is not None
+
+    if not (isByExpireTime or isByMaxReports):
+        raise HTTPException(
+            status_code=400,
+            detail="The request must contain either a maximumNumberOfReports or a monitorExpireTime",
+        )
+
+    # One time request
+    if item_in.maximumNumberOfReports == 1:
+        if item_in.monitoringType == MonitoringType.LOCATION_REPORTING:
+
+            item_in.locationArea = LocationArea(
+                cellIds=[ue.Cell_id],
```

Comment:
If a subscription is created while `Cell_id` is `None` this will fail, because `cellIds` doesn't allow `None` values.